tests/util/xcm.ts

```diff
@@ -340,7 +368,10 @@ export class XcmFragment {
   }
 
   // Add a `DepositAsset` instruction
-  deposit_asset(max_assets: bigint = 1n): this {
+  deposit_asset(
+    max_assets: bigint = 1n,
```

Comment:
This parameter not exist anymore in XCMv3, so your PR will not work. To  be able to test that, migrate to v3 at least one test that use the deposit instruction.

In XCMv3 the max number of assets should be in the filter itself( in a counted variant). So we can't support both versions at the same time, we have 2 solutions:

- create a separate method `deposit_asset_v3`
- remove support for xcm v2

Reply:
I will add a new method `deposit_asset_v3` as you suggest, to support both versions.